providers/certificate.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Author:: Blair Hamilton (<blairham@me.com>)
+# Cookbook Name:: windows
+# Provider:: certificate
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/mixin/shell_out'
+
+include Chef::Mixin::ShellOut
+
+action :add do
+  unless @current_resource.exists
+    cmd = "certutil -f -addstore #{@new_resource.name} #{@new_resource.infile}"
+    Chef::Log.debug(cmd)
+    shell_out!(cmd)
+    new_resource.updated_by_last_action true
+    Chef::Log.info("#{@new_resource.name} added to store.")
+  else
+    Chef::Log.debug("#{@new_resource.name} already exists in store - nothing to do.")
+  end
+end
+
+action :delete do
+  if @current_resource.exists
+    Chef::Application.fatal!("Required attribute cert_id not set") if (@new_resource.cert_id).nil?
+    cmd = "certutil -delstore #{@new_resource.name} #{@new_resource.cert_id}"
+    Chef::Log.debug(cmd)
+    shell_out!(cmd)
+    new_resource.updated_by_last_action true
+    Chef::Log.info("#{@new_resource.name} removed from store.")
+  else
+    Chef::Log.debug("#{@new_resource.name} does not exist in store - nothing to do.")
+  end
+end
+
+def load_current_resource
+  @current_resource = Chef::Resource::WindowsService.new(@new_resource.name)
+  @current_resource.name(@new_resource.name)
+
+  if cert_exists?(@current_resource.name)
+    @current_resource.exists = true
+  end
+end
+
+private
+
+def cert_exists?(name)
+  Chef::Log.debug "Checking to see if this cert is in the store: '#{ name }'"
+  cmd = shell_out!("certutil -verifystore #{name}", {:returns => [0,-2147024894]})
+  cmd.stderr.empty? && (cmd.stdout =~ /Certificate is valid/i)
+end
```

Comment:
So I've been looking into this and this actually verify the cert in the store, so if it's not in the store it will not return "Certificate is valid". I haven't been able to identify another way that doesn't open a GUI window. I'm open to suggestions, I do like the idea of hashing it and finding it but is seems that verifystore is already doing that.

https://technet.microsoft.com/library/cc732443.aspx#BKMK_verifystore